lib/internal/worker.js

```diff
@@ -212,9 +211,7 @@ class Worker extends EventEmitter {
     }
 
     if (!options.eval) {
-      if (!path.isAbsolute(filename)) {
-        throw new ERR_WORKER_NEED_ABSOLUTE_PATH(filename);
-      }
+      filename = path.resolve(filename);
```

Comment:
I would prefer if we can add the filter here to only apply path.resolve for absolute, `./` and `../` paths.

Reply:
On it.